Models/MicroClimate/ZoneMicroClimate.cs

```diff
@@ -80,16 +80,13 @@ private void DefineLayers()
                     double HeightMetres = Math.Round(Canopies[compNo].Canopy.Height, 5) / 1000.0; // Round off a bit and convert mm to m } }
                     double DepthMetres = Math.Round(Canopies[compNo].Canopy.Depth, 5) / 1000.0; // Round off a bit and convert mm to m } }
                     double canopyBase = HeightMetres - DepthMetres;
-                    if (Array.IndexOf(nodes, HeightMetres) == -1)
+                    if (Array.IndexOf(nodes, HeightMetres) == -1 | Array.IndexOf(nodes, canopyBase) == -1)
```

Comment:
This should be logical OR rather than the bitwise OR (note the second '|' character)

```suggestion
                    if (Array.IndexOf(nodes, HeightMetres) == -1 || Array.IndexOf(nodes, canopyBase) == -1)
```